lib/config.js

```diff
@@ -45,6 +45,14 @@ export default function subscribeConfig(conn, configChanged) {
       updateConfig({ services: Object.assign({}, config.services, { [domain]: domainInfo }) });
     }
 
+    function processServiceRemoved(event) {
+      if (config === null) return;
+
+      const { domain, service } = event.data;
+
+      updateConfig({ services: Object.assign({}, config.services, Object.assign({}, config.services.[domain], { [service]: null} )) });
```

Comment:
@balloob Would this not be more efficient?

```
function processServiceRemoved(event) {
  if (config === null) return;

  const { domain, service } = event.data;

  if (!config.services[domain] || !config.services[domain][service]) return;

  delete config.services[domain][service];
  if (!Object.keys(config.services[domain]).length)
    delete config.services[domain];

  updateConfig({});
}
```

Using `delete` does not set the value to `null`, but actually deletes the reference from memory.